docs/transient-crr-source.md

```diff
@@ -0,0 +1,177 @@
+writing the source metadata through the backbeat route, the
+replication status processor also updates the location of the object
+to be the default configured CRR target location, if it is one of the
+object's CRR targets, otherwise behavior TBD.
+
+If the update succeeds, it then publishes a message to the
+`backbeat-gc` queue to delete the transient **data**. The messages
+should contain a list of data locations to delete, in a similar format
+than in the original object's locations.
+
+##### Garbage collector
+
+We introduce a new garbage collector process, which is a kafka
+consumer of the garbage collection queue (`backbeat-gc`). It will get
+rid of the data locations defined in the messages. It may use
+sproxyd's batch delete API when deleting on RING for more efficiency.
+
+Eventually this process could be used to garbage-collect data from
+other processes (e.g. for MPU).
+
+#### Federation
+
+Note: non-zenko only
+
+For Federation deployments, we translate the `is_transient` attributes
+of location constraints into a config parameter `isTransient` in the
+cloudserver config files.
+
+#### Cloudserver
+
+Cloudserver is updated to support the new `isTransient` attribute of
+location constraints:
+
+* When receiving a PUT, if the location constraint that applies has a
+  `isTransient` config attribute set to `true`, it will set a property
+  `isTransient: true` in the `location` field of object's metadata (or
+  alternatively in another place in metadata in case `location` is not
+  a good fit), so to inform CRR that this location is meant to be
+  temporary.
+
+## Definition of API and its parameters
+
+Note: Federation deployment only (not valid for Zenko/K8s deployment)
+
+Location constraints have the optional extra boolean attribute
+`is_transient` that tells if this location is meant to be short-lived
+and turned into the CRR location configured as the default location,
+once all CRR targets are up-to-date.
+
+Example of setting in group_vars/all:
+
+```yaml
+location_constraints:
+  cloud-1:
+    type: azure
+  cloud-2:
+    type: aws_s3
+  local-ring:
+    type: scality
+    is_transient: true
+    connector:
+      sproxyd:
+        chord_cos: 3
+        bootstrap_list:
+          - node1:4244
+          - node2:4244
+          - node3:4244
+          - node4:4244
+          - node5:4244
+          - node6:4244
+```
+
+## Contract provided
+
+* Data will use space on the transient source no more time than needed
+  to sync the data to all CRR targets.
+
+* After CRR is complete, the object location should be updated to
+  point to the preferred (default) CRR target.
+
+## Useful Explanations
+
+## Dependencies
+
+The feature relies on Backbeat CRR.
+
+This means that backbeat must be enabled to use transient source.
+
+## Operational Considerations
+
+### Link with CRR
+
+This feature only works when buckets are configured with at least one
+replication target, and likely makes real sense when there are at
+least two targets.
```

Comment:
It seems like we should actually avoid using a transient location if just replicating to a sole target public cloud as it would be a superfluous use of resources. Maybe it's not worth complicating though.

Reply:
As it is implemented today, it expresses functionally as "the location is transient as long as you have some (one or more) replication targets", where replacing by "as long as you have two or more replication targets" could make the behavior more confusing.

Also there may be use cases where having a transient source doing CRR to a single public cloud could make sense, e.g. if you have a very low local storage latency and you want to benefit from this for fast local writes, and have background parallel writes to the cloud through CRR.

Reply:
In this respect I could remove the "likely makes real sense..." part.